package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -55,7 +77,102 @@ def to_json(self) -> str:
+            # we do it in two steps. Also need to add os.sep so it works with older s3fs version.
+            # This is a known bug in s3fs
+            remote_dbs = self._remote_fs.glob(f"{self.remote_location}/*.db")
+            logger.debug(
+                f"Downloading {len(remote_dbs)} remote session stores to local..."
+            )
+            for remote_db in remote_dbs:
+                self._remote_fs.get(remote_db, str(Path(self.location).parent) + os.sep)
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _merge(self):
+        """Merges all the session store databases stored at the
+        specified locations into the user's local session_store.db
+
+        Notes:
+        - This method uses multiple SQLAlchemy engines to connect to the
+        user's session_store.db and to all the other downloaded dbs.
+        - It is assumed that all the databases share the same schema.
+        - In the version 1.0 - we only merge the runs table which
+        contains all the experiments.
+        """
+
+        all_new_runs = []
+
+        with self._db_session_class() as session:
+            existing_run_ids = session.execute(select(RunModel.id)).scalars().all()
+
+        # Look at all databases in the local session store directory
+        # that aren't the actual session_store.db itself.
+        downloaded_db_locations = set(Path(self.location).parent.glob("*.db")) - {
+            Path(self.location)
+        }
+
+        logger.debug(
+            f"Checking {len(downloaded_db_locations)} downloaded session stores for new runs..."
+        )
+        for downloaded_db_location in downloaded_db_locations:
+            engine = create_engine(f"sqlite:///{downloaded_db_location}")
+            with Session(engine) as session:
+                new_runs = (
+                    session.query(RunModel)
+                    .filter(RunModel.id.not_in(existing_run_ids))
+                    .all()
+                )
+
+                existing_run_ids.extend([run.id for run in new_runs])
+                all_new_runs.extend(new_runs)
+                logger.debug(
+                    f"Found {len(new_runs)} new runs in downloaded session store"
+                    f" {downloaded_db_location.name}"
+                )
+
+        if all_new_runs:
+            logger.debug(f"Adding {len(all_new_runs)} new runs to session store...")
+            with self._db_session_class.begin() as session:
+                for run in all_new_runs:
+                    session.merge(run)
+
+    def sync(self):
+        """
+        Synchronizes the user's local session_store.db with
+        remote session_store.db stored on a cloud storage service.
+        """
+
+        if self.remote_location:
+            self._download()
+            # We don't want a failed merge to stop the whole kedro-viz process.
+            try:
+                self._merge()
+            except Exception as exc:
```

Comment:
Again I'd be more specific about where this exception is thrown, so in this case on `sync`